[BUG] 장기간 비밀번호 미변경 사용자 인증 실패 시 서버 오류로 처리되는 문제

### ❗ 문제 상황

* 일부 사용자에게서 **간헐적으로 로그인/비밀번호 재설정 실패 발생**
* 로그 상으로는 인증 서버 장애처럼 보이는 오류 발생

```text
AuthServerException [SMU001]
샘물포털 인증 서버에 일시적으로 연결할 수 없습니다.
```

* 실제 원인 분석 결과
  👉 **장기간 비밀번호를 변경하지 않은 사용자 계정**에서만 발생

---

### 🔍 원인 분석

* 샘물 포탈 인증 과정 중
  **비밀번호 만료 계정**의 경우 정상 인증 페이지가 아닌 흐름으로 응답
* 이 과정에서 `https://smsso.smu.ac.kr/Login.do` 요청이

  * `404 Not Found` 로 응답됨
* 현재 로직에서는 이를 **인증 서버 장애(AuthServerException)** 로 처리하고 있음

```text
Caused by: org.jsoup.HttpStatusException
Status=404, URL=https://smsso.smu.ac.kr/Login.do
```

➡️ **사용자 상태 문제임에도 서버 장애로 오인**되는 구조

---

### ✅ 개선 내용

* 비밀번호 장기 미변경 사용자 케이스를 **명확한 인증 오류로 분리 처리**
* 신규 에러 코드 추가

```java
AUTH_PASSWORD_EXPIRED(
    423,
    "AUTH003",
    "장기간 비밀번호를 변경하지 않으셨습니다. 샘물 포탈을 통해 비밀번호 변경 후 로그인해주시기 바랍니다."
)
```

* 의미

  * `423 Locked` : 계정은 존재하나 정책상 접근 제한 상태
  * 사용자에게 **정확한 조치 안내 제공**

---

### 🎯 기대 효과

* 인증 서버 장애(SMU001)와 사용자 상태 오류를 명확히 구분
* 불필요한 장애 오인 및 문의 감소
* 사용자 UX 개선 (즉시 해결 가능한 원인 안내)

--- a/src/main/java/com/smunity/exception/code/AuthErrorCode.java
+++ b/src/main/java/com/smunity/exception/code/AuthErrorCode.java
@@ -7,9 +7,12 @@ import lombok.RequiredArgsConstructor;
 @RequiredArgsConstructor
 public enum AuthErrorCode implements BaseCode {
 
-    // Auth Errors
+    // Client Errors
     AUTH_UNAUTHORIZED(401, "AUTH001", "학번 및 비밀번호가 일치하지 않습니다."),
     AUTH_EXCEEDED_LOGIN_ATTEMPTS(423, "AUTH002", "로그인 실패 5회 초과, 샘물 포탈을 통해 비밀번호 초기화를 진행해주시기 바랍니다."),
+    AUTH_PASSWORD_EXPIRED(423, "AUTH003", "장기간 비밀번호를 변경하지 않으셨습니다. 샘물 포탈을 통해 비밀번호 변경 후 로그인해주시기 바랍니다."),
+
+    // Server Errors
     SMU_LOGIN_FAILURE(503, "SMU001", "샘물포털 인증 서버에 일시적으로 연결할 수 없습니다. 잠시 후 다시 시도해 주세요."),
     SMU_FETCH_FAILURE(503, "SMU002", "샘물포털 서버에서 데이터를 불러올 수 없습니다. 잠시 후 다시 시도해 주세요."),
     SMU_LOGIN_TIMEOUT(504, "SMU003", "샘물포털 인증 서버 응답 시간이 초과되었습니다. 잠시 후 다시 시도해 주세요."),

--- a/src/main/java/com/smunity/util/LoginUtil.java
+++ b/src/main/java/com/smunity/util/LoginUtil.java
@@ -4,6 +4,7 @@ import com.smunity.dto.AuthRequestDto;
 import com.smunity.exception.AuthClientException;
 import com.smunity.exception.AuthServerException;
 import org.jsoup.Connection;
+import org.jsoup.HttpStatusException;
 import org.jsoup.Jsoup;
 
 import java.io.IOException;
@@ -22,6 +23,8 @@ public class LoginUtil {
         try {
             Connection.Response loginResponse = executeLogin(requestDto);
             return getSessionCookies(loginResponse);
+        } catch (HttpStatusException e) {
+            throw new AuthClientException(AUTH_PASSWORD_EXPIRED);
         } catch (SocketTimeoutException e) {
             throw new AuthServerException(SMU_LOGIN_TIMEOUT, e);
         } catch (IOException e) {

